CMakeLists.txt

```diff
@@ -95,87 +80,151 @@ add_library(
+
+target_include_directories(
+	cuda-api-wrappers
+	PUBLIC
+		"$<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/src>"
+		"$<INSTALL_INTERFACE:include>"
+)
+
+target_link_libraries(cuda-api-wrappers cudart)
+
+# To enable `target_link_library(foo cuda-api-wrappers::cuda-api-wrappers)`
+# in other CMake projects, which include this project with `add_subdirectory`
+add_library(cuda-api-wrappers::cuda-api-wrappers ALIAS cuda-api-wrappers)
+
 # -----------------------
 # Examples / Tests
 # -----------------------
 
-find_library(CUDART_LIBRARY cudart ${CMAKE_CUDA_IMPLICIT_LINK_DIRECTORIES})
-link_libraries(cuda-api-wrappers ${CUDART_LIBRARY})
-
 set(CMAKE_RUNTIME_OUTPUT_DIRECTORY "examples/bin")
+
 add_executable(vectorAdd EXCLUDE_FROM_ALL examples/modified_cuda_samples/vectorAdd/vectorAdd.cu)
+target_link_libraries(vectorAdd cuda-api-wrappers)
+
 add_executable(inlinePTX EXCLUDE_FROM_ALL examples/modified_cuda_samples/inlinePTX/inlinePTX.cu)
+target_link_libraries(inlinePTX cuda-api-wrappers)
+
 add_executable(simpleStreams EXCLUDE_FROM_ALL examples/modified_cuda_samples/simpleStreams/simpleStreams.cu)
+target_link_libraries(simpleStreams cuda-api-wrappers)
+
 add_executable(simpleIPC EXCLUDE_FROM_ALL examples/modified_cuda_samples/simpleIPC/simpleIPC.cu)
+target_link_libraries(simpleIPC cuda-api-wrappers)
+
 #----
 add_custom_target(modified_cuda_samples)
-add_dependencies(modified_cuda_samples vectorAdd inlinePTX simpleStreams simpleIPC)
+add_dependencies(
+	modified_cuda_samples
+		vectorAdd
+		inlinePTX
+		simpleStreams
+		simpleIPC
+)
 
+#----
 add_executable(version_management EXCLUDE_FROM_ALL examples/by_runtime_api_module/version_management.cpp)
-target_link_libraries(version_management  ${CUDA_LIBRARIES})
+target_link_libraries(version_management cuda-api-wrappers)
+
 add_executable(error_handling EXCLUDE_FROM_ALL examples/by_runtime_api_module/error_handling.cu)
+target_link_libraries(error_handling cuda-api-wrappers)
+
 add_executable(device_management EXCLUDE_FROM_ALL examples/by_runtime_api_module/device_management.cpp)
+target_link_libraries(device_management cuda-api-wrappers)
+
 add_executable(execution_control EXCLUDE_FROM_ALL examples/by_runtime_api_module/execution_control.cu)
+target_link_libraries(execution_control cuda-api-wrappers)
 target_compile_options(execution_control PRIVATE -rdc true)
+
 add_executable(stream_management EXCLUDE_FROM_ALL examples/by_runtime_api_module/stream_management.cu)
+target_link_libraries(stream_management cuda-api-wrappers)
+
 add_executable(event_management EXCLUDE_FROM_ALL examples/by_runtime_api_module/event_management.cu)
+target_link_libraries(event_management cuda-api-wrappers)
+
 add_executable(io_compute_overlap_with_streams EXCLUDE_FROM_ALL examples/other/io_compute_overlap_with_streams.cu)
+target_link_libraries(io_compute_overlap_with_streams cuda-api-wrappers)
+
+add_executable(ipc EXCLUDE_FROM_ALL examples/by_runtime_api_module/ipc.cpp)
```

Comment:
Oh, damn. That file is nearly empty. I shouldn't have committed it at all... anyway, it shouldn't be built now, before actual test code has actually been written in.